src/lib/hooks.ts

```diff
@@ -64,3 +65,7 @@ export function useRefresh(): () => void {
     const [, refresh] = React.useState(0);
     return React.useCallback(() => refresh(() => Math.random()), [refresh]);
 }
+
+export function useMainGoban(): GobanCore | null {
```

Comment:
This isn't really a hook yet.  As in, it won't trigger an update when global_goban changes.  Notice that the other hooks in this file are composing other hooks.

https://react.dev/learn/reusing-logic-with-custom-hooks